pkg/upgrade/upgrade_state.go

```diff
@@ -850,22 +913,16 @@ func (m *ClusterUpgradeStateManagerImpl) ProcessUncordonRequiredNodes(
 }
 
 func (m *ClusterUpgradeStateManagerImpl) isDriverPodInSync(ctx context.Context, nodeState *NodeUpgradeState) (bool, error) {
-	podRevisionHash, err := m.PodManager.GetPodControllerRevisionHash(ctx, nodeState.DriverPod)
-	if err != nil {
-		m.Log.V(consts.LogLevelError).Error(
-			err, "Failed to get pod template revision hash", "pod", nodeState.DriverPod)
-		return false, err
+	if nodeState.IsOrphanedPod() {
+		return false, nil
 	}
-	m.Log.V(consts.LogLevelDebug).Info("pod template revision hash", "hash", podRevisionHash)
-	daemonsetRevisionHash, err := m.PodManager.GetDaemonsetControllerRevisionHash(ctx, nodeState.DriverDaemonSet)
+	isPodSynced, err := m.podInSyncWithDS(ctx, nodeState)
```

Comment:
looking at these changes all the more reason to split between orphan and pod in sync with ds states

Reply:
Done